src/create_form.ts

```diff
@@ -225,10 +225,26 @@ export const createForm: CreateForm = <F extends FormValues>(
   };
 
   const reset: Reset<F> = (values, options = {}) => {
+    const currentValues = values;
     const newValues = values ? (values as F) : structuredClone(defaultValues);
 
-    setValues(() => newValues);
-    resetErrors();
+    /**
+     * checks the option keepValues to retain the 
+     * state info towards the reset option
+     */
+    const finalValues = options.keepValues 
+      ? { ...newValues, ...currentValues }
```

Comment:
shallow clone does not work with nested objects, I think you can safely use previous newValues resolution from line 229 and call setValues conditionally https://github.com/tatsmaki/solid-hook-form/pull/30/files#r2594903985